classes/class-plugin-paths.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Help resolve paths and URLs relative to plugin directories.
+ *
+ * @package Advanced_Multi_Block
+ */
+
+namespace Advanced_Multi_Block;
+
+/**
+ * Path resolver class.
+ */
+class Plugin_Paths {
+
+	/**
+	 * Absolute path to a directory inside the plugin.
+	 *
+	 * @var string
+	 */
+	private string $plugin_dir;
+
+	/**
+	 * Setup the class.
+	 *
+	 * @param string $plugin_dir Absolute path to the plugin directory.
+	 */
+	public function __construct( string $plugin_dir ) {
+		$this->plugin_dir = rtrim( $plugin_dir, '\\/' );
+	}
+
+	/**
+	 * Get a URL for a path relative to the plugin directory.
+	 *
+	 * @param string|null $relative_path Optional. Relative path from the plugin directory.
+	 *                                   Default null for the base plugin URL.
+	 *
+	 * @return string The full URL.
+	 */
+	public function get_url( ?string $relative_path = null ): string {
+		if ( isset( $relative_path ) ) {
+			return plugins_url( ltrim( $relative_path, '/' ), $this->plugin_dir . '/fake.file' );
```

Comment:
That `/fake.file` definitely feels like a hack but I'm not aware of a better helper that would accept a directory path as the argument :)